pkg/pillar/agentlog/agentlog.go

```diff
@@ -28,14 +30,38 @@ var savedAgentName string = "unknown" // Keep for signal and exit handlers
 func initImpl(agentName string, logdir string, redirect bool,
 	text bool) (*os.File, error) {
 
-	logfile := fmt.Sprintf("%s/%s.log", logdir, agentName)
-	logf, err := os.OpenFile(logfile, os.O_RDWR|os.O_CREATE|os.O_APPEND,
-		0666)
-	if err != nil {
-		return nil, err
+	var err error
+	var logf *os.File
+	var logToSyslog = false
+	if os.Getenv("LOG_TO_SYSLOG") != "" {
+		logToSyslog = true
+	}
+
+	if text {
+		logfile := fmt.Sprintf("%s/%s.log", logdir, agentName)
+		logf, err = os.OpenFile(logfile, os.O_RDWR|os.O_CREATE|os.O_APPEND,
+			0666)
+		if err != nil {
+			return nil, err
+		}
 	}
 	if redirect {
-		log.SetOutput(logf)
+		if text {
+			log.SetOutput(logf)
+		} else if logToSyslog {
+			log.SetOutput(ioutil.Discard)
+			syslogFlags := syslog.LOG_INFO | syslog.LOG_DEBUG | syslog.LOG_ERR |
+				syslog.LOG_NOTICE | syslog.LOG_WARNING | syslog.LOG_CRIT |
+				syslog.LOG_ALERT | syslog.LOG_EMERG
+			hook, err := lSyslog.NewSyslogHook("", "", syslogFlags, agentName)
+			if err == nil {
+				log.AddHook(hook)
+			} else {
+				return nil, err
+			}
+		} else {
+			log.SetOutput(os.Stdout)
```

Comment:
Since I'm working on making pillar services run as standalone containers, how can I trigger this branch to make sure all logging actually goes to stdout?

Reply:
I have tested locally by un-setting LOG_TO_SYSLOG for one service and see that service logs going to memlogd buffer. Not sure if I got your question though. Can you please elaborate?

Reply:
That's probably me being confused about text vs. redirect flags -- I'll wait for your documentation to be added to this PR.

Btw, speaking of documentation -- we need to start a separate docs/LOGGING.md documenting all the point you're making in Descriptions of your PRs and then some!

This could be a separate PR though -- no need to overload this one.

Reply:
Can you please have a look at the function comments that I have added for initImpl?

Reply:
When individual services are started stand alone logs should get printed directly on the terminal screen.